packages/amplify_api/lib/src/graphql/graphql_response_decoder.dart

```diff
@@ -0,0 +1,37 @@
+import 'dart:convert';
+
+import 'package:amplify_api/amplify_api.dart';
+
+class GraphQLResponseDecoder {
+  // Singleton methods/properties
+  // usage: GraphQLResponseDecoder.instance;
+  GraphQLResponseDecoder._();
+
+  static final GraphQLResponseDecoder _instance = GraphQLResponseDecoder._();
+
+  static GraphQLResponseDecoder get instance => _instance;
+
+  GraphQLResponse<T> decode<T>(
+      {required GraphQLRequest request,
+      required String data,
+      required List<GraphQLResponseError> errors}) {
+    // if no ModelType fallback to default
+    if (request.modelType == null) {
+      return GraphQLResponse<T>(data: data as T, errors: errors);
+    }
+
+    if (request.decodePath == null) {
+      throw ApiException('No decodePath found',
+          recoverySuggestion: 'Include decodePath when creating a request');
```

Comment:
nit: I suggest adding changing this recovery suggestion to 'Include decodePath when creating a request that includes a modelType.'